Add popular token list

danfinlay opened the pull request:
Half implements #175

Adds a tab-bar to the account-detail view that allows the display of tokens. The tokens are drawn from [eth-contract-metadata](https://github.com/MetaMask/eth-contract-metadata/blob/master/contract-map.json) for now.

Any of those tokens that the current user has a balance of will be displayed in their token list.

![screen shot 2017-06-13 at 5 04 55 pm](https://user-images.githubusercontent.com/542863/27111175-3f9c37e6-5063-11e7-82f7-387957b08eff.png)
![screen shot 2017-06-13 at 4 30 01 pm](https://user-images.githubusercontent.com/542863/27111174-3f9c3ae8-5063-11e7-92bd-3941a3aaaebc.png)

Things to do before merge:
- [x] Persist the selected token tab selection (so it is an implicit preference)
- [x] Vertically center token loading indication
- [x] Refresh view on network change.
- [x] Check what's up with the [eth-token-tracker] polling in the popup, it seems like it is not waiting the interval.

Related features not covered in this PR:
- [ ] Add ability to add tokens to the list. #1616
- [ ] Add ability to send tokens on send form. #784
- [ ] Represent token transfers in tx confirmation view. #552

danfinlay committed:
Add functional but ugly and hard-coded token list

danfinlay committed:
Add token-list dev state

danfinlay committed:
Get token list looking ok

danfinlay committed:
Add placeholder etherscan token icons

danfinlay committed:
Add missing tx manager state

danfinlay committed:
Merge branch 'master' into AddTokenList

danfinlay committed:
Fix build for eth-contract-metadata

danfinlay committed:
Auto populate token list with popular token balances

Half implements #175

Things to do:
- Add ability to add tokens to the list.
- Persist the token tab selection (so it is an implicit preference)
- Check what's up with the token-tracker polling, it seems like it is not waiting the interval.

danfinlay committed:
Add token list note to changelog

danfinlay committed:
Remove bad state file

danfinlay committed:
Rename history to sent

flyswatter commented:
Build seems broken by https://github.com/MetaMask/eth-block-tracker/issues/22

danfinlay committed:
Persist selected account tab

Also improve error handling with token balances.

danfinlay committed:
Refresh token balance on network change

danfinlay committed:
Vertically center loading indication

danfinlay committed:
Bump token-tracker version

danfinlay committed:
Add no tokens message

danfinlay committed:
Remove excessive log

danfinlay committed:
Multiple loading style improvements

- When seeking network, show a full screen loading indication + message.
- Network menu is still accessible "over" this indication.
- Top Menu-Droppo components now slide *under* the menu bar like they should.
- Loading indication opacity increased to increase message legibility.

danfinlay committed:
Hide message on normal loading

danfinlay committed:
Merge remote-tracking branch 'origin/master' into AddTokenList

flyswatter commented:
Since switching network now refreshes everything on the screen, I've also added a full-screen "Searching for network" loading indicator, to reduce visual confusion during these transitions.

![screen shot 2017-06-14 at 6 51 24 pm](https://user-images.githubusercontent.com/542863/27201170-7d61469c-51d1-11e7-9f01-08bb20f3dbe7.png)
![screen shot 2017-06-14 at 6 47 47 pm](https://user-images.githubusercontent.com/542863/27201169-7d5e1f30-51d1-11e7-842e-8ad570a3f1ed.png)

Commit:
Merge branch 'AddTokenList' of github.com:MetaMask/metamask-plugin into AddTokenList

Commit:
Merge branch 'master' into AddTokenList

danfinlay committed:
Add MetaMark support

danfinlay committed:
Merge branch 'AddTokenList' of github.com:MetaMask/metamask-plugin into AddTokenList

danfinlay committed:
Fix token balance rendering

danfinlay committed:
Add better event lifecycle management to token list.

Token list now renders errors when a token lookup fails.
Also now cleans up event listeners when re-initializing the token list.

danfinlay committed:
Bump circleCI node version to 8.0.0

danfinlay committed:
Print integration build errors
